checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java

```diff
@@ -63,19 +67,39 @@ public void exportProfile(RulesProfile profile, Writer writer) {
             final List<ActiveRule> activeRules = profile
                     .getActiveRulesByRepository(CheckstyleConstants.REPOSITORY_KEY);
             if (activeRules != null) {
-                final Map<String, List<ActiveRule>> activeRulesByConfigKey =
-                        arrangeByConfigKey(activeRules);
+                final List<ActiveRuleWrapper> activeRuleWrappers = new ArrayList<>();
+                for (ActiveRule activeRule : activeRules) {
+                    activeRuleWrappers.add(new ActiveRuleWrapperImpl(activeRule));
+                }
+                final Map<String, List<ActiveRuleWrapper>> activeRulesByConfigKey =
+                        arrangeByConfigKey(activeRuleWrappers);
                 generateXml(writer, activeRulesByConfigKey);
             }
         }
         catch (IOException ex) {
             throw new IllegalStateException("Fail to export the profile " + profile, ex);
         }
+    }
+
+    public void exportProfile(ActiveRules activeRules, Writer writer) {
+        try {
+            final List<ActiveRuleWrapper> activeRuleWrappers = new ArrayList<>();
+            for (org.sonar.api.batch.rule.ActiveRule activeRule : activeRules
```

Comment:
This class looks imported, so you don't need the full classpath.

Reply:
No, this is the new canner side `ActiveRule`, only the server side `ActiveRule` is imported.

Reply:
My bad, import had an `s` on the end. It was `ActiveRules` being imported.